src/org/labkey/test/components/ui/grids/CustomizeGridDialog.java

```diff
@@ -0,0 +1,541 @@
+    {
+        StringBuilder fieldKey = new StringBuilder();
+
+        Iterator<String> iterator = Arrays.stream(fields).iterator();
+
+        while(iterator.hasNext())
+        {
+            fieldKey.append(iterator.next().replace(" ", ""));
+
+            expandOrCollapseByFieldKey(fieldKey.toString(), true);
+
+            // If this isn't the last item in the collection don't add a "/" to the expected data-fieldkey value.
+            if(iterator.hasNext())
+            {
+                fieldKey.append("/");
+            }
+
+        }
+
+        return this;
+    }
+
+    /**
+     * Collapse a top level field in 'Available Fields' panel.
+     *
+     * @param fieldName Name of the field to collapse.
+     * @return This dialog.
+     */
+    public CustomizeGridDialog collapseAvailableField(String fieldName)
+    {
+        expandOrCollapseByFieldKey(fieldName.replace(" ", ""), false);
+        return this;
+    }
+
+    /**
+     * Private helper function that will expand or collapse a row in the 'Available Fields' panel.
+     *
+     * @param fieldKey The data-fieldkey value of the field to expand.
+     * @param expand True to expand false to collapse.
+     */
+    private void expandOrCollapseByFieldKey(String fieldKey, boolean expand)
+    {
+        WebElement listItem = elementCache().getListItemElementByFieldKey(fieldKey);
+
+        Assert.assertTrue(String.format("Field with data-fieldkey attribute '%s' is not visible in the 'Available Fields' panel.", fieldKey),
+                listItem.isDisplayed());
+
+        String iconClass = expand ? "fa-plus-square" : "fa-minus-square";
+
+        WebElement expandIcon = Locator.tagWithClass("div", "field-expand-icon")
+                .withChild(Locator.tagWithClass("i", iconClass))
+                .findElement(listItem);
+
+        String errorMessage;
+
+        if(expand)
+            errorMessage = String.format("There is no expand icon for field with data-fieldkey attribute '%s' in the 'Available Fields' panel.", fieldKey);
+        else
+            errorMessage = String.format("There is no collapse icon for field with data-fieldkey attribute '%s' in the 'Available Fields' panel.", fieldKey);
+
+        Assert.assertTrue(errorMessage, listItem.isDisplayed());
+
+        expandIcon.click();
+
+    }
+
+    /**
+     * Private helper to see if the row with the given data-fieldkey value has a checkmark or not.
+     *
+     * @param fieldKey The data-fieldkey value of the row/field to check.
+     * @return True if row is expanded.
+     */
+    private boolean isFieldKeyExpanded(String fieldKey)
+    {
+        WebElement listItem = elementCache().getListItemElementByFieldKey(fieldKey);
+
+        // As long as there is no plus/expand icon then this field is expanded.
+        return   !Locator.tagWithClass("div", "field-expand-icon")
+                .withChild(Locator.tagWithClass("i", "fa-plus-square"))
+                .findWhenNeeded(listItem).isDisplayed();
+    }
```

Comment:
This should probably take a `WebElement` instead of the fieldKey. Most usages of it have already found the list item before calling this. Also, fail if the field isn't expandable.
```suggestion
    /**
     * Private helper to see if the given row is expanded or not. Will throw if the row is not expandable.
     *
     * @param listItem The available field list item to inspect.
     * @return True if row is expanded.
     */
    private boolean isFieldKeyExpanded(WebElement listItem)
    {
        WebElement expandIcon = Locator.byClass("div", "field-expand-icon").childTag("i").findElement(listItem);
        return expandIcon.getAttribute("class").contains("fa-minus-square");
    }
```

Reply:
This is only used in `expandAvailableFields`.  Taking a field-key value is very explicit, it is an attribute on a specific element. Taking a WebElement is equivalent to passing in an object and is more susceptible to error (wrongly scoped element is passed in).